stdlib/public/core/Collection.swift

```diff
@@ -673,6 +677,7 @@ public protocol MutableCollection : MutableIndexable, Collection {
 }
 
 extension MutableCollection {
+  @discardableResult
```

Comment:
Please drop this one, for the same reason as you explained on `withUnsafeMutableBufferPointer`.